Dockerfile

```diff
@@ -143,8 +143,8 @@ ENV PATH="/home/ubuntu/.yarn/bin:/home/ubuntu/.config/yarn/global/node_modules/.
 COPY package.json .
 COPY yarn.lock .
 RUN yarn install --production && yarn cache clean
-COPY lib lib
+COPY dist dist
```

Comment:
I think this may cause a problem with our automated Docker builds. I need to work out if Docker can actually build the project first before building the Dockerfile. I suspect it can't, which means the Docker images would break.

Reply:
OK, can we use 2 stage docker build?

Reply:
then we can in first stage build renovate with node:lts-alpine and then copy dist from stage=0

Reply:
I was thinking like this:
![image](https://user-images.githubusercontent.com/6311784/56791494-78aa5780-6807-11e9-938b-920baa1760e3.png)
From: https://docs.docker.com/docker-hub/builds/advanced/

But your idea might be simpler. Let's build on the same image/platform we use though.

Reply:
if we use 2 stages we do not have the dev deps installed in the final layer :-)

so i would prefer 2 stages, but we can do it in a single stage too. docker builds supports multistage builds :-)

Reply:
I will update the dockerfile, then you can see it :-)

Reply:
ok thanks. I *think* I am nearly done, just in case you haven't started yet

Reply:
OK, i'm done 😄

Reply:
```patch
diff --git a/Dockerfile b/Dockerfile
index a9800f62..02f35410 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,3 +1,16 @@
+FROM node:lts-alpine@sha256:313c5c88acc0ec12a9abca9a83719065dfd54e94ffc56464b7ce24998dd2838d
+
+WORKDIR /usr/src/app/
+
+COPY package.json .
+COPY yarn.lock .
+RUN yarn install
+
+COPY lib lib
+
+RUN yarn build
+
+
 FROM amd64/ubuntu:18.04@sha256:f2557f94cac1cc4509d0483cb6e302da841ecd6f82eb2e91dc7ba6cfd0c580ab

 LABEL maintainer="Rhys Arkins <rhys@arkins.net>"
@@ -143,7 +156,7 @@ ENV PATH="/home/ubuntu/.yarn/bin:/home/ubuntu/.config/yarn/global/node_modules/.
 COPY package.json .
 COPY yarn.lock .
 RUN yarn install --production && yarn cache clean
-COPY dist dist
+COPY --from=0 /usr/src/app/dist dist
 COPY bin bin

 ENTRYPOINT ["node", "/usr/src/app/dist/renovate.js"]
```

Reply:
why we need this:
https://github.com/renovatebot/renovate/blob/ba543a029512a26c17f6437f488c73e1d0a389dd/Dockerfile#L107

Reply:
I forget, but it looks like the type of line not to remove unless you do know why it'd needed :D

Reply:
Potentially chowning it to the ubuntu user would also solve whatever problem that caused it